cairo-contracts/packages/core/src/channel/components/handler.cairo

```diff
@@ -1078,6 +1089,15 @@ pub mod ChannelHandlerComponent {
                     root_on_b
                 );
         }
+
+        fn assert_authorized_relayer(self: @ComponentState<TContractState>) {
+            let client_comp = get_dep_component!(self, ClientHandler);
+
+            assert(
+                client_comp.in_allowed_relayers(get_caller_address()),
+                ClientErrors::UNAUTHORIZED_RELAYER
+            );
```

Comment:
I don't like that the `allowed_relayer` list is stored in client contract :sweat_smile:

But I can't think of any clean solution either. Anyway, we will remove these once we have the proper light client. So, this is just a remark.

Reply:
I see, no way for now. Specially as it's a temporary solution.